pkg/helmshim/helm.go

```diff
@@ -32,6 +33,19 @@ func MustRenderChart(opts ...RenderOption) {
 	}
 }
 
+// isNullObject checks if the given unstructured object is equivalent to an empty K8S object.
+// This is used when then input helm chart includes an empty target (for example: empty yaml file with comments).
+func isNullObject(o *unstructured.Unstructured) bool {
+	if o == nil {
+		return true
+	}
+	b, err := json.Marshal(o)
```

Comment:
I'd prefer to avoid the overhead of json encoding here. Would the test pass with `o == nil || o.Object == nil`?

Reply:
yes and i think these two checks should handle 99% of the cases. But becuase the unstructured.Unstructured is using a customized JSON marshaler: https://github.com/kubernetes/apimachinery/blob/955939ffb8198b47fe11e4e53121d529ee6f946b/pkg/apis/meta/v1/unstructured/unstructured.go#L124 therefore i want to avoid some edge cases that we are not aware.

Reply:
agreed - best to hedge against changes to the marshaler since we will hit the branch on line 42 most of the time anyway